src/Theme.cpp

```diff
@@ -0,0 +1,236 @@
+	theme.text[Theme::Logic]                = readFormat(settings, "logic", baseTheme.text[Theme::Logic]);
+	theme.text[Theme::NonPrintingCharacter] = readFormat(settings, "non_printing_character", baseTheme.text[Theme::NonPrintingCharacter]);
+	theme.text[Theme::Operator]             = readFormat(settings, "operator", baseTheme.text[Theme::Operator]);
+	theme.text[Theme::Prefix]               = readFormat(settings, "prefix", baseTheme.text[Theme::Prefix]);
+	theme.text[Theme::Ptr]                  = readFormat(settings, "ptr", baseTheme.text[Theme::Ptr]);
+	theme.text[Theme::Register]             = readFormat(settings, "register", baseTheme.text[Theme::Register]);
+	theme.text[Theme::Shift]                = readFormat(settings, "shift", baseTheme.text[Theme::Shift]);
+	theme.text[Theme::Stack]                = readFormat(settings, "stack", baseTheme.text[Theme::Stack]);
+	theme.text[Theme::System]               = readFormat(settings, "system", baseTheme.text[Theme::System]);
+	theme.text[Theme::TakenJump]            = readFormat(settings, "taken_jump", baseTheme.text[Theme::TakenJump]);
+
+	// misc settings
+	theme.misc[Theme::Badge] = readFormat(settings, "badge", baseTheme.misc[Theme::Badge]);
+
+	settings.endGroup();
+
+	return theme;
+}
+
+/**
+ * @brief readSystemTheme
+ * @return
+ */
+Theme readSystemTheme() {
+	if (QApplication::palette().window().color().lightnessF() >= 0.5) {
+		QSettings settings(":/themes/system-light.ini", QSettings::IniFormat);
+		return readTheme(settings);
+	} else {
+		QSettings settings(":/themes/system-dark.ini", QSettings::IniFormat);
+		return readTheme(settings);
+	}
+}
+
+/**
+ * @brief readTheme
+ * @return
+ */
+Theme readTheme() {
+
+	// Ensure that the directory exists, this will help users know where to put the theme files
+	QDir().mkpath(themeDirectory());
+
+	Theme system = readSystemTheme();
+
+	QString theme_name = edb::v1::config().theme_name;
+
+	// Handle the built-in themese
+	if(theme_name == "System") {
+		return system;
+	} else if(theme_name == "Dark [Built-in]") {
+		QSettings settings(":/themes/dark.ini", QSettings::IniFormat);
+		return readTheme(settings, system);
+	} else if (theme_name == "Light [Built-in]") {
+		QSettings settings(":/themes/light.ini", QSettings::IniFormat);
+		return readTheme(settings, system);
+	}
+
+	QString themeFile = themeDirectory() + QDir::separator() + theme_name;
+	QSettings settings(themeFile, QSettings::IniFormat);
+	return readTheme(settings, system);
+}
+
+}
+
+/**
+ * @brief Theme::load
+ * @return
+ */
+Theme Theme::load() {
+
+	// we do this function indirection to make it so the theme is
+	// read once even if this function is called several times
+	static Theme theme = readTheme();
+	return theme;
+}
+
+
+/**
+ * @brief Theme::userThemes
+ * @return
+ */
+QStringList Theme::userThemes() {
+	QDir directory(themeDirectory());
+	return directory.entryList(QStringList() << "*.ini", QDir::Files);
```

Comment:
`QStringList` has a constructor that takes `std::initializer_list<QString>`.

Reply:
I'm not sure that Qt 5.2 had that constructor though. Will have to wait for version bump.

Reply:
The docs say the constructor appeared in Qt 4.8.